scripts/setup_irods.py

```diff
@@ -111,12 +111,10 @@ def setup_server(irods_config, json_configuration_file=None, test_mode=False):
             from irods import database_interface
             if database_interface.database_already_in_use_by_irods(irods_config):
                 raise IrodsError('Database specified already in use by iRODS.')
-        # default resource
+        # default resource information
+        # XXX This is now required at the top level of the JSON object. Should this be optional?
```

Comment:
i think it has to be defined - this is an unattended setup - so don't want any magic.

Reply:
I believe the setup script provides defaults for this depending on the server's role.

Do we want this to follow the same pattern? For example, if they don't provide a resource name and they are configuring a provider, default to demoResc.

Reply:
We can also improve this in a later release.

Reply:
Well... not ideal - but it seems okay.    Especially if that's what already happens.

No magic should be the end goal I think.

Reply:
Ok. I won't waste time trying to change this too much. I'll make sure it works and we can talk about how much magic setup should have later.